lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -351,20 +391,24 @@ def add_copy_resources_script_phase
             end
             return
           end
+
           script_path = target.copy_resources_script_relative_path
-          input_paths = []
-          output_paths = []
+          input_paths_by_config = {}
+          output_paths_by_config = {}
           unless installation_options.disable_input_output_paths?
-            resource_paths_by_config = target.resource_paths_by_config
-            resource_paths_flattened = resource_paths_by_config.values.flatten.uniq
-            input_paths = [target.copy_resources_script_relative_path, *resource_paths_flattened]
-            output_paths = TargetIntegrator.resource_output_paths(resource_paths_flattened)
+            target.resource_paths_by_config.each do |config, resource_paths|
+              input_paths_key = [target.copy_resources_script_input_files_path(config), target.copy_resources_script_input_files_relative_path]
+              input_paths_by_config[input_paths_key] = [script_path] + resource_paths
+
+              output_paths_key = [target.copy_resources_script_output_files_path(config), target.copy_resources_script_output_files_relative_path]
+              output_paths_by_config[output_paths_key] = TargetIntegrator.resource_output_paths(resource_paths)
```

Comment:
could the key be modeled by a `Struct` or something similar? using an Array as a hash key feels a bit confusing to me but I know that is a thing in Ruby land

Reply:
It could be